packages/bot/src/scratch/index.js

```diff
@@ -209,13 +209,17 @@ class DBot {
         this.valueInputLimitationsListener({}, true);
 
         const all_blocks  = this.workspace.getAllBlocks(true);
-        const error_block = all_blocks.find(block => block.is_error_highlighted && !block.disabled);
-
-        if (error_block) {
-            const { run_panel } = ScratchStore.instance.root_store;
-            const message       = localize('The block(s) highlighted in red are missing input values. Please update them and click "Run bot".');
-            this.workspace.centerOnBlock(error_block.id);
-            run_panel.showErrorMessage(message);
+        const error_blocks = all_blocks
+            .filter(block => block.is_error_highlighted && !block.disabled)
+            .filter((block, index, self) => index === self.findIndex(b => b.error_message === block.error_message));
+
+        if (error_blocks.length) {
+            error_blocks.forEach(block => {
+                const { run_panel } = ScratchStore.instance.root_store;
+                const message       = block.error_message;
+                this.workspace.centerOnBlock(block.id);
+                run_panel.showErrorMessage(message);
+            });
```

Comment:
There is no need to center on each block, it will always center on the last block according to this logic. I think it makes more sense to center on the first block that errored.